Initial Draft of publiccode.yml

cklellie opened the pull request:
In order to publish to the European Open Source Software Catalogue we need to be registered with offentligkod.se and have a publiccode.yml file.

An initial version of the file has been created, it is a work in progress with this PR serving as a mechanism to review and update the contents of the publiccode.yml appropriately before it is merged.

The specification for publiccode.yml is available here:
https://yml.publiccode.tools/index.html
https://github.com/publiccodeyml/publiccode.yml

There are also github actions and validators to assist with lifecycle management of this file.

Prior to the merge this PR the file should be check with a validation tool to ensure it is well formed.

lazan commented:
@OlofGullnas can you check with Bolagsverket that it's ok to refer to them as users of `casual`

lazan commented on publiccode.yml:
```diff
@@ -0,0 +1,79 @@
+
+usedBy: #optional
+  - Försäkringskassan
+  - Bolagsverket
```
```yaml
 - Swedish social insurance system
 - Swedish Companies Registration Office
 ```

cklellie replied:
Social Insurance Agency i have seen before in the FK site in english.
https://www.forsakringskassan.se/english/moving-to-working-studying-or-newly-arrived-in-sweden/social-insurance-in-sweden
Also, should we have the actual names in brackets a bit like in the links on the side of that site?
```yaml
- Swedish Social Insurance Agency (Försäkringskassan)
- Swedish Companies Registration Office (Bolagsverket)
```

lazan replied:
I found it here: https://www.forsakringskassan.se/english

cklellie replied:
I see. But that is the system, rather than the company perhaps, on the previous link i provided in the first paragraph it seemed to describe it like that too.
Bit more digging... most things are agencies from what I can see.
https://www.government.se/government-agencies/social-insurance-agency--forsakringskassan/
https://www.government.se/government-agencies/swedish-pensions-agency-pensionsmyndigheten/
Though bolagsverket is an office :)
https://www.government.se/government-agencies/swedish-companies-registration-office/

lazan replied:
Maybe there's a distinction since they're self financed.

lazan commented on publiccode.yml:
```diff
@@ -0,0 +1,79 @@
+
+indendedAudience: #optional ??? restricted list: https://yml.publiccode.tools/scope-list.html#valid-tags
+  scope:
+    - welfare
```
We should have every tag.
```yaml
- agriculture
- culture
- defence
- education
- emergency-services
- employment
- energy
- environment
- finance-and-economic-development
- foreign-affairs
- government
- healthcare
- infrastructures
- justice
- local-authorities
- manufacturing
- research
- science-and-technology
- security
- society
- sport
- tourism
- transportation
- welfare
```

lazan commented on publiccode.yml:
```diff
@@ -0,0 +1,79 @@
+publiccodeYmlVersion: "0.4"
+
+name: Casual
```
Lower case `c` :). `casual`.

cklellie replied:
of course, sorry! :)

lazan commented:
FK is ok with it.

cklellie commented:
I have updated after your initial comments @lazan

If you have a look around here at other examples:
https://interoperable-europe.ec.europa.eu/eu-oss-catalogue
For example here:
https://interoperable-europe.ec.europa.eu/eu-oss-catalogue/solutions/opencity-italia-la-stanza-del-cittadino

"version" and "releaseDate" are quite prominate on the pages, should we also add this, even if it does mean a bit of overhead with releases.
Could we also get an `svg` logo to place in the repo which we can reference?

I also found an example for Design System from FK here:

https://github.com/Forsakringskassan/designsystem/blob/main/publiccode.yml

OlofGullnas commented:
Í checked with Patrik Westman some time ago, and Bolagsverket is also OK with being mentioned/Olof

lazan commented:
> I have updated after your initial comments @lazan
> 
> If you have a look around here at other examples: https://interoperable-europe.ec.europa.eu/eu-oss-catalogue For example here: https://interoperable-europe.ec.europa.eu/eu-oss-catalogue/solutions/opencity-italia-la-stanza-del-cittadino
> 
> "version" and "releaseDate" are quite prominate on the pages, should we also add this, even if it does mean a bit of overhead with releases. 

I would imagine that it's not super critical to keep this updated?
 
> Could we also get an `svg` logo to place in the repo which we can reference?

Doesn't it work with `png`? I don't want to add it to 1.7 if we've not got any other changes.

> 
> I also found an example for Design System from FK here:
> 
> https://github.com/Forsakringskassan/designsystem/blob/main/publiccode.yml

lazan commented on publiccode.yml:
```diff
@@ -0,0 +1,102 @@
+maintenance:
+  type: "community"
+
+  contacts:
```
Me and @ahhud 

-  Fredrik Eriksson
-  casual@laz.se
- Hans Bergkvist
-  hans.bergkvist@ymail.com

lazan replied:
Or more better... Why not just have:

```yaml
    contacts:
        name: community
        email: https://discord.com/channels/960208635458363425/960208766157070366 
        
 ```

cklellie replied:
No, they don't want you to do that.
https://yml.publiccode.tools/schema.core.html#contact
Must be "real" people and not mailing lists.
I'll add your initial suggestion.

cklellie commented:
> > I have updated after your initial comments @lazan
> > If you have a look around here at other examples: https://interoperable-europe.ec.europa.eu/eu-oss-catalogue For example here: https://interoperable-europe.ec.europa.eu/eu-oss-catalogue/solutions/opencity-italia-la-stanza-del-cittadino
> > "version" and "releaseDate" are quite prominate on the pages, should we also add this, even if it does mean a bit of overhead with releases.
> 
> I would imagine that it's not super critical to keep this updated?
> 
> > Could we also get an `svg` logo to place in the repo which we can reference?
> 
> Doesn't it work with `png`? I don't want to add it to 1.7 if we've not got any other changes.
> 
> > I also found an example for Design System from FK here:
> > https://github.com/Forsakringskassan/designsystem/blob/main/publiccode.yml

The version should just be with, i think they mentioned using it within their metrics somehow related to how old something was or similar but i forget exactly. But no, if we just set it, this should be fine.

The icon, yes it will work, but svg was the preference.

cklellie committed:
documentation: enable indexing by EU Open Source Software Catalogue

In order to publish to the European Open Source Software Catalogue
we need to be registered with offentligkod.se and have a
publiccode.yml file.

The specification for publiccode.yml is available here:
https://yml.publiccode.tools/index.html
https://github.com/publiccodeyml/publiccode.yml
Editor/ Validation tool for publiccode.yml is available here:
https://publiccode-editor.developers.italia.it/

lazan reviewed (changes requested):
Change the commit message to something like:

```git
documentation: added publiccode.yaml for eu exposure 

This  patch adds  publiccode.yaml  to enable indexing 
in EU Open Source Solutions Catalogue
bla bla bla.
```
Every line is max 72 characters.

cklellie commented:
> Change the commit message to something like:
> 
> ```
> documentation: added publiccode.yaml for eu exposure 
> 
> This  patch adds  publiccode.yaml  to enable indexing 
> in EU Open Source Solutions Catalogue
> bla bla bla.
> ```
> 
> Every line is max 72 characters.

done

lazan reviewed (approved):
Nice!
